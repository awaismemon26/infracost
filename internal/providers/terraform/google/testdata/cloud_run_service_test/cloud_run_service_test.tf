provider "google" {
  credentials = "{\"type\":\"service_account\"}"
  project     = "my-project"
  region      = "europe-west3"
}

resource "google_cloud_run_service" "throttling_enabled" {
  name     = "cloud-run-service-test"
  location = "europe-west3"
  template {
    spec {
      containers {
        image = "us-docker.pkg.dev/cloudrun/container/hello"
        resources {
          limits = {
            "cpu"    = "1"
            "memory" = "512Mi"
          }
        }
      }
    }
  }
  traffic {
    percent         = 100
    latest_revision = true
  }
}
resource "google_cloud_run_service" "throttling_disabled" {
  name     = "cloud-run-service-test"
  location = "europe-west3"
  template {
    spec {
      containers {
        image = "us-docker.pkg.dev/cloudrun/container/hello"
      }
    }
  }
  metadata {
    annotations = {
      "run.googleapis.com/cpu-throttling": false
      "autoscaling.knative.dev/minScale": "1"
    }
  }
  traffic {
    percent         = 100
    latest_revision = true
  }
}
