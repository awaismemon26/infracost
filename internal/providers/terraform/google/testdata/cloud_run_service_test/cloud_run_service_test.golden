
 Name                                          Monthly Qty  Unit          Monthly Cost 
                                                                                       
 google_cloud_run_service.throttling_disabled                                          
 ├─ CPU Allocation Time (Always-on)              1,314,000  vCPU-seconds        $28.38 
 └─ Memory Allocation Time (Always-on)             657,000  GiB-seconds          $1.58 
                                                                                       
 google_cloud_run_service.throttling_enabled                                           
 ├─ CPU Allocation Time (tier 2)                 1,800,000  vCPU-seconds        $60.48 
 ├─ Memory Allocation Time (tier 2)                900,000  GiB-seconds          $3.15 
 └─ Number of requests                             500,000  request              $0.20 
                                                                                       
 OVERALL TOTAL                                                                  $93.79 
──────────────────────────────────
2 cloud resources were detected:
∙ 2 were estimated

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Project                                            ┃ Monthly cost ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━━┫
┃ TestCloudRunService                                ┃ $94          ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━┛